### PR TITLE
Add initial metadata support, Improve support for client timezones

### PR DESCRIPTION
**Pull request by solflare**
This adds initial support for a metadata query param for report URLs that is an encoded json payload that currently supports a single timezone string value: `timezone`. The timezone value allows the report the have the context of the client's timezone when filtering datetime values. This works under the assumption that the datetime values that are being compared from the DB are in UTC

### CODE REVIEW

File: resources/views/base-web.blade.php [hunk 1]
```diff
@@ -105,6 +105,19 @@ function popupConfirm(row, title, body, routeTemplate, routeReplacements, routeH
             })
         }
 
+        function addMetaDataToUrl(url)
```
Comment (solflare) [inline resources/views/base-web.blade.php:108]: I kept this simple for now but I was thinking we could do something more config based e.g. 'use-client-timezone'. I didn't want to commit too heavily on an implementation before getting thoughts

Another thing to think about: if users implement their own frontend for the data these reports provide then they will need to implement a similar function to get the same behavior

File: src/BaseFeatures/Filters/GreaterThanFilter.php [hunk 2]
```diff
@@ -15,8 +17,21 @@ class GreaterThanFilter extends BaseFilter
     public function apply(Builder $builder, array $options = []) : Builder
     {
         $action = $this->getAction();
+        $value = $this->getValue();
 
-        return $builder->$action($this->getField(), '>', $this->getValue());
+        if ($this->valueIsDate()) {
```
Comment (solflare) [inline src/BaseFeatures/Filters/GreaterThanFilter.php]: I was wondering if it was maybe worth separating out the Date oriented filters from the others but didn't want to prematurely optimize

Comment (qschmick) [reply src/BaseFeatures/Filters/GreaterThanFilter.php]: Why not pass `$options` into the `getValue` method? Then do the conversion in there

File: resources/views/base-web.blade.php (hunk 1 above)
Comment (qschmick) [reply resources/views/base-web.blade.php:108]: I like the idea of this morphing into config based.

Agreed, users would need to implement sending this data if they wanted to but at that point they are doing custom UI. These views are publishable so they already would have the existing JS to duplicate or extend.

**Pull request merged**